src/nvim/file_search.c

```diff
@@ -1522,6 +1523,35 @@ find_file_in_path_option (
   return file_name;
 }
 
+static void apply_autocmd_dirchanged(char_u *new_dir, CdScope scope)
```

Comment:
Maybe `do_autocmd_dirchanged` is a better name. The precedent is `do_autocmd_event()` and `yank_do_autocmd`. The latter should probably be renamed to `do_autocmd_textyankpost`.